### PR TITLE
feat(server): Creating Server with a Handle WIP

### PR DESCRIPTION
**Pull request by Thomspoon**
- [x] The commit messages match the guidelines in https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md#git-commit-guidelines

This PR is a work-in-progress for #1322.

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -376,7 +403,7 @@ impl<T, B> Service for HttpService<T>
     }
 }
 
-impl<S, B> Server<S, B>
+impl<'a, S, B> Server<S, B>
```
Comment (Thomspoon) [inline src/server/mod.rs]: Oops. That wasn't supposed to be in there.

File: src/server/mod.rs [hunk 2]
```diff
@@ -389,7 +416,10 @@ impl<S, B> Server<S, B>
     /// Returns a handle to the underlying event loop that this server will be
     /// running on.
     pub fn handle(&self) -> Handle {
-        self.core.handle()
+        match self.internal {
+            ServerInternal::Core(ref core) => core.handle().clone(),
```
Comment (seanmonstar) [inline src/server/mod.rs]: The clone here isn't need, `core.handle()` already returns `Handle`.

File: src/server/mod.rs [hunk 3]
```diff
@@ -105,13 +112,33 @@ impl<B: AsRef<[u8]> + 'static> Http<B> {
...
+    /// Bind the provided `addr` and return a server that has
+    /// a shared `Core`.
+    /// 
+    /// This method, when provided a handle to a `Core`, allows
+    /// the ability to bind multiple servers to the same `Core`.
+    pub fn bind_handle<S, Bd>(&self, addr: &SocketAddr, new_service: S, handle: Handle) -> Server<S, Bd>
+        where S: NewService<Request = Request, Response = Response<Bd>, Error = ::Error> + 'static,
+            Bd: Stream<Item=B, Error=::Error>,
+    {
+        let listener = TcpListener::bind(addr, &handle).unwrap();
```
Comment (seanmonstar) [inline src/server/mod.rs]: This probably means `bind_handle` should return a `Result`, just like `bind` does, so that we don't panic.

**Comment by seanmonstar** [comment]
Master has grown to have a `Serve` type which will be used to explore replacing `Server`. `Serve` doesn't require a `Core`, but instead is a stream of binding connections with HTTP.

Thanks for taking a stab at this!

**Pull request closed without merging**